app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,252 @@
+        @DisplayName("만약 tasks가 비어있다면")
+        class Context_of_empty_tasks {
+
+            @Test
+            @DisplayName("비어있는 배열을 반환한다")
+            void it_returns_empty_array() {
+                List<Task> tasks = taskService.getTasks();
+
+                assertThat(tasks).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 tasks가 비어있지 않다면")
+        class Context_of_tasks_not_empty {
+
+            private Task task1;
+            private Task task2;
+            private Long currentTaskId = 0L;
+
+            @BeforeEach
+            void append_two_task_to_tasks() {
+                currentTaskId += 1;
+                task1 = generateTask(currentTaskId, "task" + currentTaskId);
+                currentTaskId += 1;
+                task2 = generateTask(currentTaskId, "task" + currentTaskId);
+
+                taskService.createTask(task1);
+                taskService.createTask(task2);
+            }
+
+            @Test
+            @DisplayName("tasks에 포함된 모든 객체를 반환한다")
+            void it_returns_all_tasks() {
+                List<Task> tasks = taskService.getTasks();
+
+                assertThat(tasks).containsExactly(task1, task2);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_of_getTask {
+
+        private Task task1;
+        private Task task2;
+        private Long currentTaskId = 0L;
+
+        @BeforeEach
+        void append_two_task_to_tasks() {
+            currentTaskId += 1;
+            task1 = generateTask(currentTaskId, "task" + currentTaskId);
+            currentTaskId += 1;
+            task2 = generateTask(currentTaskId, "task" + currentTaskId);
+
+            taskService.createTask(task1);
+            taskService.createTask(task2);
+        }
+
+        @Nested
+        @DisplayName("만약 tasks에 존재하지 않는 task id를 인자로 입력하면")
+        class Context_of_invalid_id {
+
+            private long invalidId;
+
+            @BeforeEach
+            void setInvalidId() {
+                invalidId = currentTaskId + 42L;
+            }
+
+            @Test
+            @DisplayName("TaskNotFoundException을 던진다")
+            void it_throws_task_not_found_exception() {
+                assertThrows(
+                        TaskNotFoundException.class,
+                        () -> taskService.getTask(invalidId));
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 tasks에 포함된 task의 id를 인자로 전달 받으면")
+        class Context_of_valid_id {
+
+            @Test
+            @DisplayName("id에 해당하는 task를 반환한다")
+            void getTask() {
+                Task task = taskService.getTask(task1.getId());
+
+                assertThat(task).isEqualTo(task1);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_of_createTask {
+
+        @Nested
+        @DisplayName("만약 Task 객체를 인자로 전달받으면")
+        class Context_of_given_task {
+
+            private Task source;
```

Comment:
실험의 핵심이 되는 `task`에 대해 다음과 같이 이름을 지어주면 어떨까요? `given~`을 보면 실험의 주요 조건이라 파악할 수 있을 것입니다.
```suggestion
            private Task givenTask;
```

Reply:
오오 테스트 코드에 더 적합한 네이밍이네요! 제안 감사합니다 :)